ci/workflows/main-cron.yml

```diff
@@ -7,7 +7,7 @@ auto-retry: &auto-retry
 steps:
   - label: "build"
     command: "ci/scripts/build.sh -p ci-release"
-    if: (!build.pull_request.labels includes "ci/main-cron/skip-ci" || build.pull_request.labels includes "ci/run-build")
+    if: (!(build.pull_request.labels includes "ci/main-cron/skip-ci" || build.env("CI_STEPS") != null) || build.pull_request.labels includes "ci/run-build" || build.env("CI_STEPS") =~ /(^|,)build(,|$$)/)
```

Comment:
Why do we need both `build.env("CI_STEPS") != null)` and `build.env("CI_STEPS") =~ /(^|,)build(,|$$)/)`

Reply:
Seems only the latter is needed.

Reply:
```suggestion
    if: ((!build.pull_request.labels includes "ci/main-cron/skip-ci" && build.env("CI_STEPS") == null) || build.pull_request.labels includes "ci/run-build" || build.env("CI_STEPS") =~ /(^|,)build(,|$$)/)
```

Reply:
It's better to expand the `!` here. Otherwise it seems confusing to me.
Additionally I think we could also have multiline conditionals:
https://buildkite.com/docs/pipelines/conditionals.
![Uploading Screenshot 2023-11-20 at 12.49.27 PM.png…]()